src/OrchardCore.Modules/OrchardCore.Secrets/Controllers/AdminController.cs

```diff
@@ -0,0 +1,360 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Localization;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.Extensions.Localization;
+using Microsoft.Extensions.Options;
+using OrchardCore.Admin;
+using OrchardCore.DisplayManagement;
+using OrchardCore.DisplayManagement.ModelBinding;
+using OrchardCore.DisplayManagement.Notify;
+using OrchardCore.Mvc.Utilities;
+using OrchardCore.Navigation;
+using OrchardCore.Routing;
+using OrchardCore.Secrets.Models;
+using OrchardCore.Secrets.Options;
+using OrchardCore.Secrets.ViewModels;
+using OrchardCore.Settings;
+
+namespace OrchardCore.Secrets.Controllers;
+
+[Admin]
+public class AdminController : Controller
+{
+    private readonly IAuthorizationService _authorizationService;
+    private readonly ISecretService _secretService;
+    private readonly IDisplayManager<SecretBase> _displayManager;
+    private readonly IUpdateModelAccessor _updateModelAccessor;
+    private readonly SecretOptions _secretsOptions;
+    private readonly ISiteService _siteService;
+    private readonly INotifier _notifier;
+
+    protected readonly dynamic New;
```

Comment:
There are some outdated `dynamic`s here.

Reply:
dynamic new is still used in 35 places, but yes better to use the factory directly, was too lazy.

Reply:
Also shapes are stored as `dynamic` below.

Reply:
Yes as you I'm not a fan of dynamic, but here it is as it is done in other places in the source code.

So maybe a separate PR to remove the usage of dynamic shapes everywhere as much as possible.

Reply:
I _am_ a fan of dynamic, but not here :). Removing it in other places is a separate issue indeed, but this is a new controller that uses an old approach of managing shapes.

Reply:
Okay, will update it